pkg/signal/trap_linux_test.go

```diff
@@ -0,0 +1,99 @@
+// +build linux
+
+package signal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTrap(t *testing.T) {
+	sigmap := map[string]os.Signal{
+		"TERM": syscall.SIGTERM,
+		"QUIT": syscall.SIGQUIT,
+		"INT":  os.Interrupt,
+	}
+
+	if os.Getenv("TEST_TRAP") == "1" {
+		defer time.Sleep(5 * time.Second)
+
+		Trap(func() {
+			time.Sleep(1 * time.Second)
```

Comment:
Does it need to sleep in the trap? Could we sleep for a few milliseconds instead of a full second?